test/OpenSpaceToolkit/Astrodynamics/Flight/Profile/Tabulated.test.cpp

```diff
@@ -0,0 +1,198 @@
+/// Apache License 2.0
+
+#include <OpenSpaceToolkit/Astrodynamics/Flight/Profile/Model/Tabulated.hpp>
+#include <OpenSpaceToolkit/Astrodynamics/Trajectory/State.hpp>
+
+#include <Global.test.hpp>
+
+using ostk::core::container::Array;
+using ostk::core::container::String;
+
+using ostk::mathematics::object::Vector3d;
+using ostk::mathematics::object::VectorXd;
+using ostk::mathematics::geometry::d3::transformation::rotation::Quaternion;
+
+using ostk::physics::time::Interval;
+using ostk::physics::time::Instant;
+using ostk::physics::time::DateTime;
+using ostk::physics::time::Scale;
+using ostk::physics::coordinate::Position;
+using ostk::physics::coordinate::Velocity;
+using ostk::physics::coordinate::Frame;
+
+using ostk::astro::trajectory::State;
+using ostk::astro::flight::profile::models::Tabulated;
+
+class OpenSpaceToolkit_Astrodynamics_Flight_Profile_Models_Tabulated : public ::testing::Test
+{
+   protected:
+    void SetUp() override
+    {
+        {
+            const Instant instant = Instant::DateTime(DateTime(2024, 1, 29, 0, 0, 0), Scale::UTC);
+            const Position position =
+                Position::Meters({755972.142139276024, -3390511.949699319433, 5955672.751532567665}, Frame::GCRF());
+            const Velocity velocity =
+                Velocity::MetersPerSecond({-563.764594800880, -6619.592151780337, -3685.668514834143}, Frame::GCRF());
+            const Quaternion attitude = {
+                -0.638160707740, -0.163520830523, 0.726693549038, 0.194751982966, Quaternion::Format::XYZS
+            };
+            const Vector3d angularVelocity = {0.0, 0.0, 0.0};
+            states_.add({instant, position, velocity, attitude, angularVelocity, Frame::GCRF()});
+        }
+
+        {
+            const Instant instant = Instant::DateTime(DateTime(2024, 1, 29, 0, 0, 30), Scale::UTC);
+            const Position position =
+                Position::Meters({738650.069144404028, -3587215.226667015813, 5841866.765430007130}, Frame::GCRF());
+            const Velocity velocity =
+                Velocity::MetersPerSecond({-590.931279522597, -6492.770161294917, -3900.690931064452}, Frame::GCRF());
+            const Quaternion attitude = {
+                -0.635371915567, -0.174049521049, 0.723374654124, 0.206732684649, Quaternion::Format::XYZS
+            };
+            const Vector3d angularVelocity = {0.0, 0.0, 0.0};
+            states_.add({instant, position, velocity, attitude, angularVelocity, Frame::GCRF()});
+        }
+    }
+
+    Array<State> states_ = Array<State>::Empty();
```

Comment:
This should work but feel free to ignore if it doesn't
```suggestion
    Array<State> states_ = Array<State>::Empty();
    Tabulated tabulated_;
```